components/python/importlib_metadata/Makefile

```diff
@@ -0,0 +1,36 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# This file was automatically generated using the following command:
+#   $WS_TOOLS/python-integrate-project importlib_metadata
+#
+
+BUILD_STYLE = setup.py
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME =		importlib_metadata
+COMPONENT_VERSION =		4.10.1
+COMPONENT_SUMMARY =		importlib_metadata - Read metadata from Python packages
+COMPONENT_PROJECT_URL =		https://github.com/python/importlib_metadata
+COMPONENT_ARCHIVE_URL =		\
+	https://files.pythonhosted.org/packages/f8/41/8ffb059708359ea14a3ec74a99a2bf0cd44a0c983a0c480d9eb7a69438bb/importlib_metadata-4.10.1.tar.gz
+COMPONENT_ARCHIVE_HASH =	\
```

Comment:
With the correct COMPONENT_ARCHIVE we can use COMPONENT_ARCHIVE_URL=  $(call pypi_url) for most python packages.

Reply:
Yes, I know. It is also the default now: https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/make-rules/setup.py-defaults.mk#L22

The `$(call pypi_url)` expands to something like `pypi:///importlib_metadata==4.10.1` which is something recognized by `userland-fetch`, but otherwise I doubt it is generally unknown.  The problem is that the `COMPONENT_ARCHIVE_URL` value could appear in man pages (see any man page generated from oi-userland, for example `man gtar`) and if we put something like `pypi:///importlib_metadata==4.10.1` there we could confuse users.  I would instead prefer to see real download url there.  OTOH, I do not know if we generate any man page from python projects, so maybe this is completely pointless.

Also, the `COMPONENT_ARCHIVE_URL` line in the `Makefile` is automatically generated so to have the full download path there causes zero additional maintenance costs.

Anyway, if you prefer the url in form of `pypi:///...` instead of `https://...` then it would be easy to achieve by modifying the `python-integrate-project` script.

Reply:
I haven't seen any package yet with such generated man pages. You are right that this would be irritating. On the other hand using pypi typically doesn't break download url's when a project moves. I don't care which one to use but using pypi looks more convenient.